src/dash-adapter.js

```diff
@@ -1283,6 +1288,27 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     });
   }
 
+  /**
+   * An handler to shaka emsg event
+   * @function _onEmsg
+   * @param {any} event - the emsg event
+   * @returns {void}
+   * @private
+   */
+  _onEmsg(event: any): void {
+    const {detail, type} = event;
+    let metadataTrack = Array.from(this._videoElement.textTracks).find(track => track.label === type);
+    if (!metadataTrack) {
+      metadataTrack = this._videoElement.addTextTrack(TextTrack.KIND.METADATA, type);
```

Comment:
are we gonna have few metadata tracks?

Reply:
yes. this is why `cuePointManager.getAllCuePoints` needs to over on all of them . here https://github.com/kaltura/kaltura-player-js/pull/509/files#diff-d14bbbfd7b8451b73314e22e98f15a1f16d1f379672902aee021c38b719b2a84R38